src/__tests__/deployments.helper.ts

```diff
@@ -96,9 +101,20 @@ export function getVersionsByChainIdByDeploymentMap(): VersionsByChainIdByDeploy
       }
 
       // Add the version to the map
-      for (const chainId of Object.keys(deployment.networkAddresses)) {
-        versionsByDeploymentByChainId[name][chainId] ??= [];
-        versionsByDeploymentByChainId[name][chainId]?.push(deployment.version);
+      if (typeof deployment.networkAddresses === 'object') {
+        const networkAddresses: Record<string, string> =
```

Comment:
We're checking whether `deployment.networkAddresses` is an `object` so it can't be `Record<string, string>` here. We need to be careful with directly declaring variable types - it can "override" `any`s.

I think the `typeof xyz === 'object` can also return `true` for `Date` objects as well but I don't think we need concern ourselves about that here.

Reply:
I've uploaded a version using just casting on this file. Do you think we should explicitly check the deployments' JSON types?